Assets/Scripts/Characters/Humans/Hero.cs

```diff
@@ -449,6 +450,12 @@ public void Update()
                 bool isLeftHookPressed;
                 BufferUpdate();
                 UpdateExt();
+                if (state != HumanState.ChangeBlade && weaponDisabledOnReloading)
+                {
+                    //If the reload animation is cancelled before the weapon has a chance to be reenabled, call this function to do that
+                    Equipment.Weapon.EnableWeapons();
+                }
```

Comment:
The weapon will be enabled repeatedly since Equipment.Weapon.EnableWeapons() does not change weaponDisabledOnReloading to false. Indeed, I don't see that weaponDisabledOnReloading is ever false. See the previous review, it might be a possible solution.

Reply:
This function is called only when its any state but reloading. I.e. if we switch to a hook animation, and the weapons are still disabled, then it will be called, which does the "reloading" instantly. It's set to false in the function that throws the blades so that its true in the frames between the blades being thrown (since they disappear then) and the blades being brought of the holsters (they are reenabled here)

Reply:
Wait